### PR TITLE
fix(823): [2] fixed not to create event when there is no job triggered by '~tag' or '~release'

### PR DESCRIPTION
**Pull request by sakka2**
## Context
Now, tag or release events are created even if user does not have '~tag' or '~release' definition in screwdriver.yaml.

## Objective
These are empty events so we should not create them.

## References
https://github.com/screwdriver-cd/screwdriver/issues/823#issuecomment-489999734

## OtherPRs
[1] https://github.com/screwdriver-cd/data-schema/pull/338

### CODE REVIEW

File: plugins/webhooks/index.js [hunk 1]
```diff
@@ -600,6 +600,15 @@ async function createEvents(eventFactory, userFactory, pipelineFactory,
...
+        const checkStartFroms = ['~tag', '~release'];
+
+        if (checkStartFroms.includes(startFrom) && !hasTriggeredJob(p, startFrom)) {
+            winston.info(
+                `event is not created because there are no jobs triggered by ${startFrom}`
+            );
+            continue; // eslint-disable-line no-continue
+        }
+
         events.push(eventFactory.create(eventConfig));
```
Comment (wdstar) [inline plugins/webhooks/index.js]: This `evens.push()` should be included in the last `if ... else ...` block.

File: plugins/webhooks/index.js [hunk 2]
```diff
@@ -600,6 +600,15 @@ async function createEvents(eventFactory, userFactory, pipelineFactory,
         await updateAdmins(userFactory, username, scmContext, p);
         /* eslint-enable no-await-in-loop */
 
+        const checkStartFroms = ['~tag', '~release'];
```
Comment (tk3fftk) [inline plugins/webhooks/index.js]: I think this information should not be in this line. This should be constant as a global variable or should be in data-schema.

Comment (tk3fftk) [reply plugins/webhooks/index.js]: Also this param name should have more information. How about `excludedStartFroms` or `excludableStartFroms`?

File: plugins/webhooks/index.js (hunk 1 above)
Comment (sakka2) [reply plugins/webhooks/index.js]: Thanks! I fixed to use `if .. else ..` block.

File: plugins/webhooks/index.js (hunk 2 above)
Comment (sakka2) [reply plugins/webhooks/index.js]: Thanks! I fixed to use data-schema.
https://github.com/screwdriver-cd/data-schema/pull/338

File: plugins/webhooks/index.js [hunk 3]
```diff
@@ -4,11 +4,14 @@ const boom = require('boom');
...
 const winston = require('winston');
 const workflowParser = require('screwdriver-workflow-parser');
+const schema = require('screwdriver-data-schema');
 
 const ANNOT_NS = 'screwdriver.cd';
 const ANNOT_CHAIN_PR = `${ANNOT_NS}/chainPR`;
 const ANNOT_RESTRICT_PR = `${ANNOT_NS}/restrictPR`;
-const CHECKOUT_URL_SCHEMA = require('screwdriver-data-schema').config.regex.CHECKOUT_URL;
+const EXTRA_TRIGGERS = schema.config.regex.EXTRA_TRIGGERS;
+const EXTRA_TRIGGERS_REGEXP = new RegExp(EXTRA_TRIGGERS);
```
Comment (wdstar) [inline plugins/webhooks/index.js]: `schema.config.regex.EXTRA_TRIGGERS` is already a regex object.
This line's  instantiation is not necessary?

Comment (sakka2) [reply plugins/webhooks/index.js]: Thanks! I fixed.

File: plugins/webhooks/index.js (hunk 2 above)
Comment (tk3fftk) [reply plugins/webhooks/index.js]: `EXTRA_TRIGGERS` looks good 👍

**Comment by sakka2** [comment]
Sorry, I fixed just the typo. Could you merge this please?

**Pull request merged**